sympy/logic/boolalg.py

```diff
@@ -679,6 +679,25 @@ def _new_args_filter(cls, args):
             newargs.append(x)
         return LatticeOp._new_args_filter(newargs, And)
 
+    def _eval_subs(self, old, new):
+        args = []
+        bad = None
+        for i in self.args:
+            try:
+                i = i.subs(old, new)
+            except:  # store any problem
```

Comment:
I would read that as "hide any problem" :)

Bare except shouldn't be used anywhere in SymPy. Catching Exception is preferable to bare except (since it won't catch e.g. KeyboardInterrupt) but is still bad.

What exception is being caught here? I guess it's TypeError (invalid nan comparison). Probably we should change it so that we have an InvalidComparisonError to be caught in these situations though.

Reply:
#17537

Reply:
Ah I see. It's still better to catch Exception rather than use bare except for that though.

In the case where this does short-circuit the exception will be suppressed and we need to be more specific about which exceptions to suppress if we're doing that. I would suggest catching only the exceptions you expect to see: anything else could be a bug somewhere.